pkg/apis/core/validation/validation.go

```diff
@@ -421,6 +421,32 @@ func ValidateVolumes(volumes []core.Volume, podMeta *metav1.ObjectMeta, fldPath
 	return vols, allErrs
 }
 
+func ValidateVolumesUpdate(volumes, oldVolumes []core.Volume, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	mapOldVolumeNames := make(map[string]core.Volume)
+	for _, oldVol := range oldVolumes {
+		o := oldVol.DeepCopy()
+		if o.VolumeSource.Ephemeral == nil {
+			continue
+		}
+		// If the existing volumes have limits set, then we skip the validation.
+		if _, ok := o.VolumeSource.Ephemeral.VolumeClaimTemplate.Spec.Resources.Limits[core.ResourceStorage]; ok {
+			return allErrs
```

Comment:
continue rather than return here? Then we'd skip any old volumes that have limits, but will verify that for any old volume that does not have a limit, that limit isn't set in the update.

Reply:
Yeah, that makes sense. Fixed.